src/qae/initial-properties.js

```diff
@@ -74,6 +74,43 @@ const properties = {
   components: [],
 };
 
+/**
+ * Extends `NxDimension`, see Engine API: `NxDimension`
+ * @typedef {object} DimensionProperties
+ * @extends NxDimension
+ * @property {InlineDimensionDef} qDef
+ */
+
+/**
+ * Extends `NxMeasure`, see Engine API: `NxMeasure`
+ * @typedef {object} MeasureProperties
+ * @extends NxMeasure
+ * @property {InlineMeasureDef} qDef
+ */
+
+/**
+ * Extends `NxInlineDimensionDef`, see Engine API: `NxInlineDimensionDef`.
+ * @typedef {object} InlineDimensionDef
+ * @extends NxInlineDimensionDef
+ * @property {ColumnWidth=} columnWidth
+ */
+
+/**
+ * Extends `NxInlineMeasureDef`, see Engine API: `NxInlineMeasureDef`.
+ * @typedef {object} InlineMeasureDef
+ * @extends NxInlineMeasureDef
+ * @property {ColumnWidth=} columnWidth
+ */
+
+/**
+ * Column width info. For the left grid, the properties are always applied.
+ * For the right grid, only the leaf nodes will listen to the properties, and the columns above will get the width of the leaves accumulated
+ * @typedef {object} ColumnWidth
+ * @property {('auto' | 'FitToContent' | 'pixels' | 'percentage')} type - Defines how the column width is set. For the right grid, `auto` calculates the width(s) so the total width of the columns equals the right grid width. If the width reaches a minimum value, the columns will overflow. For the left grid, `auto` is N/A and defaults to `fitToContent`. `fitToContent` calculates a width based on the cells' content. `pixels` uses a specified pixel value. `percentage` sets the column width to specified percentage of the chart width
```

Comment:
`cells'` should maybe be `cells`

Reply:
no but it should be `cell's`, good catch

Reply:
actually it should probably be `column's`